yash-datalake/azuredeploy.json

```diff
@@ -0,0 +1,576 @@
+          },
+          "eventhub_namespace": {
+            "value": "[parameters('eventHubNamespaceName')]"
+          },
+          "adla_account_name": {
+            "value": "[parameters('adlAnalyticsName')]"
+          },
+          "DataCleaningStreamAnalyticsJob": {
+            "value": "[variables('DataCleaningStreamAnalyticsJob')]"
+          },
+          "DataAggregatorStreamAnalyticsJob1": {
+            "value": "[variables('DataAggregatorStreamAnalyticsJob1')]"
+          },
+          "DataAggregatorStreamAnalyticsJob2": {
+            "value": "[variables('DataAggregatorStreamAnalyticsJob2')]"
+          },
+          "DataAggregatorStreamAnalyticsJob3": {
+            "value": "[variables('DataAggregatorStreamAnalyticsJob3')]"
+          },
+          "_artifactsLocation": {
+            "value": "[parameters('_artifactsLocation')]"
+          },
+          "_artifactsLocationSasToken": {
+            "value": "[parameters('_artifactsLocationSasToken')]"
+          }
+        }
+      }
+    },
+    {
+      "type": "Microsoft.Resources/deployments",
+      "name": "KibanaESDeployedVM",
+      "apiVersion": "2018-02-01",
+      "dependsOn": [
+        "storageAccount"
+      ],
+      "properties": {
+        "mode": "Incremental",
+        "templateLink": {
+          "uri": "[uri(parameters('_artifactsLocation'),concat('nested/elasticSearchKibana.json', parameters('_artifactsLocationSasToken')))]",
+          "contentVersion": "1.0.0.0"
+        },
+        "parameters": {
+          "vmSize": {
+            "value": "[parameters('vmSize')]"
+          },
+          "username": {
+            "value": "[parameters('vm_username')]"
+          },
+          "password": {
+            "value": "[parameters('vm_password')]"
+          },
+          "dnsLabelPrefix": {
+            "value": "[variables('dnsLabelPrefix')]"
+          },
+          "location": {
+            "value": "[parameters('location')]"
+          },
+          "_artifactsLocation": {
+            "value": "[parameters('_artifactsLocation')]"
+          },
+          "_artifactsLocationSasToken": {
+            "value": "[parameters('_artifactsLocationSasToken')]"
+          }
+        }
+      }
+    },
+    {
+      "type": "Microsoft.Resources/deployments",
+      "name": "KibanaFunctionApp",
+      "apiVersion": "2018-02-01",
+      "dependsOn": [
+        "KibanaESDeployedVM"
+      ],
+      "properties": {
+        "mode": "Incremental",
+        "templateLink": {
+          "uri": "[uri(parameters('_artifactsLocation'),concat('nested/blobTriggerdFunction.json', parameters('_artifactsLocationSasToken')))]",
+          "contentVersion": "1.0.0.0"
+        },
+        "parameters": {
+          "storage_name": {
+            "value": "[variables('storageAccountName')]"
+          },
+          "kibanaFunctionHost": {
+            "value": "[Concat(variables('ES_Function_name'),'.azurewebsites.net')]"

```

Comment:
use reference() to get the endpoint

Reply:
Changed function logic because reference cannot be used here since this resource is created here only. So cannot use reference to refer the hostname.